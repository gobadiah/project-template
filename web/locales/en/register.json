{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
