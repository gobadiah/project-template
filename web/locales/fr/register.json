{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
