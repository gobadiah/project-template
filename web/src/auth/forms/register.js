import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from '~/components/forms/fields';
import { Form } from '~/components/forms';

import { PureComponent } from '~/components/base';

class RegisterForm extends PureComponent {
  render() {
    return (
      <Form {...this.props}>
        <EmailField key='email' />
        <PasswordField
          key='password'
          autoComplete='new-password'
        />
        <PasswordField
          key='confirm_password'
          name='confirm_password'
          label='Password confirmation'
          autoComplete='new-password'
        />
        <FirstNameField key='first_name' />
        <LastNameField key='last_name' />
        <BirthdayField key='birthday' />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
