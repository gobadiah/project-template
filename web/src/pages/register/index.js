import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
