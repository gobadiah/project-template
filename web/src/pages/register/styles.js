import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
