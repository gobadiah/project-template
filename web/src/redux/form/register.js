import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
