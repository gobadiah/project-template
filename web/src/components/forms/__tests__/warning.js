import 'jsdom-global/register';

import React from 'react';
import { mount, shallow } from 'enzyme';

import Warning from '../warning';

const context = {
  i18n: {
    on: jest.fn(),
  },
  t: x => x,
};

describe('Warning', () => {
  it('should match snapshot when mounting', () => {
    const tree = mount(<Warning warning={['One warning', 'Two warning']} />, { context });
    expect(tree).toMatchSnapshot();
  });

  it('shallow render should contain elements', () => {
    const wrapper = shallow(<Warning warning='This is an warning' />, { context });
    expect(wrapper.contains('This is an warning')).toBeTruthy();
  });
});
