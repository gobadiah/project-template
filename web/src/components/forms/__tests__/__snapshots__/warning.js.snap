// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning should match snapshot when mounting 1`] = `
<Warning
  warning={
    Array [
      "One warning",
      "Two warning",
    ]
  }
>
  <div
    key="0"
  >
    One warning
  </div>
  <div
    key="1"
  >
    Two warning
  </div>
</Warning>
`;
